spring-restdocs-core/src/main/java/org/springframework/restdocs/payload/ContentTypeHandlerFactory.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2014-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.payload;
+
+import org.springframework.http.MediaType;
+
+/**
+ * Factory providing access to {@link ContentHandler} implementation for the content type.
+ *
+ * @author Mathias Düsterhöft
+ */
+final class ContentTypeHandlerFactory {
```

Comment:
I think this could be replaced with a static `forContent` method on `ContentHandler`.

Reply:
Are you sure? - `ContentHandler` is an interface at the moment. So we could use a default method. Then the factory method feels a little odd when working with the implementations. 

That is why I introduced the factory class. 

We could also make `ContentHandler` an abstract class.

Let me know which way you prefer.